py/util.h

```diff
@@ -44,14 +44,19 @@ convert_to_double( PyObject* obj, double& out )
 inline bool
 convert_to_strength( PyObject* value, double& out )
 {
+  std::string str;
+
 #if PY_MAJOR_VERSION >= 3
     if( PyUnicode_Check( value ) )
     {
-        std::string str( PyUnicode_AsUTF8( value ) );
+      str = PyUnicode_AsUTF8( value );
 #else
-    if( PyString_Check( value ) )
+    if( PyString_Check( value ) | PyUnicode_Check( value ))
```

Comment:
I would make this and `if { } else if {} else {}` block instead of nested `if`

Reply:
I agree it looks a bit inelegant, and I'm not the most elegant programmer, so no offense if you or someone else changes it.   But, you need an outer `if` to check for a string to send to the string translators below, and then you need to decide which string decoder you want.

Reply:
I am not sure it would make this any more elegant as we would have either to duplicate the following logic (string value comparison) or refactor it in a function.  To me this looks fine.

Reply:
Yep. You're both right. I only looked at the diff preview and was missing the extra context.